designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,544 @@
+1. Creating configuration files will be more complicated.
+1. The `--print-config` option becomes less useful because we can't output objects in a meaningful way.
+1. People depending on environments may find this change particularly painful.
+
+## Backwards Compatibility Analysis
+
+The intent of this proposal is to replace the current `.eslintrc` format, but can be implemented incrementally so as to cause as little disturbance to ESLint users as possible
+
+In the first phase, I envision this:
+
+1. `.eslint.js` can be implemented alongside `.eslintrc`.
+1. If `.eslint.js` is found, then:
+    1. All `.eslintrc` files are ignored.
+    1. `.eslintignore` is ignored.
+    1. `--ignore-file` is ignored.
+    1. `--rulesdir` is ignored.
+    1. `--env` is ignored.
+    1. `eslint-env` config comments are ignored.
+1. If `.eslint.js` is not found, then fall back to the current behavior.
+
+This keeps the current behavior for the majority of users while allowing some users to test out the new functionality. Also, `-c` could not be used with `.eslint.js` in this phase.
+
+In the second phase (and in a major release), ESLint will emit deprecation warnings whenever the original functionality is used but will still honor them so long as `.eslint.js` is not found.
+
+In the third phase (and in another major release), the original functionality is removed.
+
+So while this is intended to be a breaking change, it will be introduced over the course of three major releases in order to give users ample time to transition.
+
+## Alternatives
+
+While there are no alternatives that cover all of the functionality in this RFC, there are alternatives designed to address various parts.
+
+* Both https://github.com/eslint/rfcs/pull/7 and https://github.com/eslint/rfcs/pull/5 specify an alternative method for resolving plugin locations in configs. This attempts to solve the problem with bundling plugins with configs (https://github.com/eslint/eslint/issues/3458). These proposals have the benefit of working within the current configuration system and requiring very few changes from users.
+* It is possible to come up with a solution to https://github.com/eslint/eslint/issues/8813 that makes use of `extends`, though this would get a bit complicated if an extended config in an `overrides` section also has `overrides`.
+* We could switch the current configuration system over so that all config files are considered to implicitly have `root:true`. This could dramatically simplify configuration searching and merging.
+
+## Open Questions
+
+1. Should `ignore` allow line breaks so it can emulate `.eslintignore`?
+1. Is `ruledefs` a create enough key name?
+1. Do we need a command line flag to opt-in to `.eslint.js` instead of trying to do it alongside the existing configuration system?
+1. Does the file pattern system actually remove the need for `--ext`?
```

Comment:
How would this affect the `eslintConfig` and `eslintIgnore` `package.json` keys? Would they be deprecated/removed in the same manner as `.eslintrc.*` files?

Reply:
Yes. The intent is that all configuration now lives in just one place.